### DISCUSSION TITLE
Swipe in favorites and routes

### DISCUSSION THREAD

**Discussion by DolomitesTom** [devemux86/cruiser#469] [Q&A] [answered]
Could you please make long press availability in favourites and routes?
Longpress on a route in my routes loads it and long press on a favorite selects it on the map, mainly used functions.

**Comment by devemux86** [comment]
@DolomitesTom do you mean long press on the map or in their lists?

- Long press on the map on favorites or routes is already available.

- Favorite and route lists have a menu button at the right with several actions.

**Reply by DolomitesTom** [reply]
I mean in the lists. Yes I know the three dots. But this is not so useful, small and not good to hit. Therefore I would like to have the most used default function directly available by longpress. Since this is not used before, I guess this could be useful for everybody.

**Reply by devemux86** [reply]
Long press is currently used for drag and reorder the items in the list.
(together with drag the left icon)

**Reply by DolomitesTom** [reply]
Didn't use this before. But when I long press at any other position, nothing happens. Why not use this for load the route?

**Reply by devemux86** [reply]
@DolomitesTom without an extra question dialog, this could confuse users.

They could press a little longer a list row and will see suddenly the map displaying a favorite or a different route.

**Comment by DolomitesTom** [comment]
OK, what about a wipe right to load and the wipe left to delete? At the moment wiping left and right would delete the route.

**Reply by WolfgangZ42** [reply]
Do we really need this Tinder behaviour just to save a single tap?

**Reply by DolomitesTom** [reply]
yes we need

**Reply by devemux86** [reply]
Something like this might work.

Although then all other lists will have swipe removal in both directions.

**Reply by Leeb64** [reply]
I'm afraid this could easily lead to misoperation...(delete)

**Reply by devemux86** [reply]
App should have similar behavior in all similar interfaces.

If we make the remove with only left swipe, then it should be done everywhere.
And let the right swipe working in the lists we want extra action.

Although, when right swipe does not have action, its animation probably cannot be disabled.

**Reply by kds112** [reply]
Maybe introduce a global setting "enable/disable swipes"? 🤔

**Reply by devemux86** [reply]
Swipe is needed to remove items in the lists.

**Reply by DolomitesTom** [reply]
Even swipe delete on the left prevents from deleting unintendedly when swipe right has a different action. You find no App where swiping left and right has the same function.

When swiping to the left unintendedly then you just don't need to save this.

I like the idea of having swiping right as context action.

**Reply by MotoUKRider** [reply]
I regard "swipe" as a dismiss gesture rather than a further action. A long-press is best suited for bringing-up a new dialog/menu, with a single-press being the "confirm/OK" gesture.

It's not like we're swiping images in a side-bar to open into a large viewing area :thinking:

**Reply by DolomitesTom** [reply]
Emux didn't like Long press. In outlook you can also configure the swipe left and right action.

Swipe= dismiss? also used Tinder too often? I have never used such social media scrap, so swiping  for me could be a quick action which I can use with thick motorcycle gloves to perform the most used navigation feature, just a practical thing.

**Reply by devemux86** [reply]
Long press without displaying a menu or confirmation dialog is not very safe, it can lead to wrong actions.

Swipe left or right at least needs more gesture work by the user, so it is a more predetermined movement.

**Comment by devemux86** [comment]
The swipe actions in all lists are:
- Left swipe to remove in all screens

- **Groups** of favorites / routes / tracks: right to open group

- **Favorites**: right to select on map

- **Routes**: right to load route

- **Tracks**: right to calculate route

- **Waypoints**: right to select on map

**Reply by devemux86** [reply]
The above described different swipe left (remove) or right (action) in lists is available for testing in:
- https://github.com/devemux86/cruiser/discussions/482

**Reply by DolomitesTom** [reply]
Works great but I would prefer when delete a shaping point that then it's deleted directly. The Button now is big enough to hit. I don't need a question are you really sure? This with a small confirm button. This is the style break. If hit accidentally, one could take it back with the back button. Only the back button is not available in this new menu.

**Reply by devemux86** [reply]
> when delete a shaping point that then it's deleted directly

We can do this. As "Undo" is always available, it is easy to revert any route changes.

**Reply by bios4** [reply]
I really like the swipe functions, more than I thought I would... 😁

> Works great but I would prefer when delete a shaping point that then it's deleted directly. [...] I don't need a question are you really sure?

When swiping left on a shaping point in the waypoints list, it is deleted (as soon as the waypoints list is saved/closed) w/o any confirmation question...

So where/during which operation do you get a question whether you are sure to delete a shaping point @DolomitesTom?

**Reply by devemux86** [reply]
> So where/during which operation do you get a question whether you are sure to delete a shaping point @DolomitesTom?

It is the delete operation in the long press (on map) menu of waypoints.

**Reply by bios4** [reply]
Yeah, but what has this to do with the swiping functions discussed here?

**Reply by DolomitesTom** [reply]
Swipe right in the waypoint list and you see the delete button

**Reply by bios4** [reply]
Not on my phone...

Swiping right centers the map on the selected waypoint and opens its menu (and yes, there also is the delete button, among other options).

Swiping left removes the selected waypoint from the waypoints list and subsequently also deletes it from the route, when the list is saved & closed.

I like the behaviour!

**Reply by devemux86** [reply]
> Swiping right centers rhe map on the selected waypoint and opens its menu (and yes, there also is the delete button, among other options).

Both of you mean the same thing! :slightly_smiling_face:

**Comment by devemux86** [comment, marked as answer]
Implemented in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

For instructions see [above](https://github.com/devemux86/cruiser/discussions/469#discussioncomment-8879031).
